### PR TITLE
Fix ApparentHorizon dependencies

### LINKED ISSUE #5244: Move StrahlkorperGr to PointwiseFunctions
From #3561. StrahlkorperGr isn't well placed in `ApparentHorizons`. The suggestion is to move it to `PointwiseFunctions/GeneralRelativity/` in a new subdirectory+library, maybe `Surfaces` or something like that.

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

This PR does several things, but the main point is that it removes the `ApparentHorizon` dependency from the `ParallelInterpolation` library

1. Move StrahlkorperGr (in `ApparentHorizons`) to `PointwiseFunctions/GR/Surfaces/` and split up into different files
2. Move TimeDerivStrahlkorper to StrahlkorperFunctions in `SphericalHarmonics`
3. Move StrahlkorperInDifferentFrame to `Domain` (it depends on `Domain`)
4. Move all AH stuff (target/callbacks) out of `ParallelInterpolation` and into `ApparentHorizons` (and have `ApparentHorizons` now depend on `ParallelInterpolation` instead of the other way around)
5. Move lots of tags from `ApparentHorizons` to other places better suited for the tags (`PointwiseFunctions/GR/Surfaces`)
6. Remove unneded dependency on `ParallelInterpolation`

This PR can be split into smaller ones if need be.

Fixes #5244

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/PointwiseFunctions/GeneralRelativity/Surfaces/EuclidianAreaElement.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace StrahlkorperGr {
+template <typename Frame>
+void euclidean_area_element(
```
Comment (nilsvu) [inline src/PointwiseFunctions/GeneralRelativity/Surfaces/EuclidianAreaElement.cpp]: Move into `AreaElement.*pp` if that's not too much work

File: src/PointwiseFunctions/GeneralRelativity/Surfaces/EuclidianSurfaceIntegralOfVector.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// Distributed under the MIT License.
```
Comment (nilsvu) [inline src/PointwiseFunctions/GeneralRelativity/Surfaces/EuclidianSurfaceIntegralOfVector.cpp]: Remove `Euclidean` from name as well I think

File: src/ApparentHorizons/CMakeLists.txt [hunk 3]
```diff
@@ -25,6 +26,7 @@ spectre_target_headers(
   ObserveCenters.hpp
   Tags.hpp
   TagsDeclarations.hpp
+  Target.hpp
```
Comment (nilsvu) [inline src/ApparentHorizons/CMakeLists.txt]: `InterpolationTarget`?

**Review by nilsvu — COMMENTED** [review]
Thanks for moving this stuff around! You know that good deeds never go unpunished, so would it be too much to ask also to move `ApparentHorizons/` -> `ParallelAlgorithms/ApparentHorizonFinder/`? 😅

**Comment by nilsvu** [comment]
Making the changes above in extra commits is fine if it's too annoying to squash those changes in.

File: src/PointwiseFunctions/GeneralRelativity/Surfaces/Tags.hpp [hunk 4]
```diff
@@ -0,0 +1,545 @@
...
+/// @{
+/// Tag to compute the time derivative of the coefficients of a Strahlkorper
+/// from a number of previous Strahlkorpers.
+template <typename Frame>
+struct TimeDerivStrahlkorper : db::SimpleTag {
+  using type = ::Strahlkorper<Frame>;
+};
+
+template <typename Frame>
+struct TimeDerivStrahlkorperCompute : db::ComputeTag,
```
Comment (nilsvu) [inline src/PointwiseFunctions/GeneralRelativity/Surfaces/Tags.hpp]: Did you intend to move this to NumericalAlgorithms/SphericalHarmonics where you also moved the time derivative function? I think the function and the tags would be best placed in the same file, perhaps `NumericalAlgorithms/SphericalHarmonics/TimeDerivStrahlkorper.*pp`.

**Review by nilsvu — COMMENTED** [review]
More cleanups that we can do later:

- [ ] Rename the `StrahlkorperGr` lib and namespace to `gr::surfaces`.
- [ ] Rename the `StrahlkorperTags` namespace to `ylm::Tags`.
- [ ] Rename the `StrahlkorperFunctions` namespace to just `ylm`.
- [ ] Move everything else in `NumericalAlgorithms/SphericalHarmonics/` into the `ylm` namespace.
- [ ] Move tags from the giant `GR/Surfaces/Tags.hpp` file and from `GR/Surfaces/ComputeItems.hpp` into the individual files that hold the corresponding function, like `RicciScalar.hpp` etc.

**Comment by knelli2** [comment]
@nilsvu Added 4 new commits that address your comments

- Rename `EuclidianSurfaceIntegralOfVector` -> `SurfaceIntegralOfVector`
- Move `euclidian_area_element` to `AreaElement` and remove `EuclidianAreaElement`
- Rename `ApparentHorizons/Target.?pp` -> `ApparentHorizons/InterpolationTarget.?pp`
- Move `TimeDerivStrahlkorper` tag from `GR/Surfaces/Tags.hpp` to `SphericalHarmonics/Tags.hpp`

Didn't bother making `TimeDerivStrahlkorper` its own file at the moment because we have to make a new file for all tags in `SphericalHarmonics/Tags.hpp` and `GR/Surfaces/Tags.hpp` later on. I'll just do it then. The rest of the changes I'll make in another PR.

**Review by nilsvu — APPROVED** [review]
Thanks!

**Pull request merged**